components/tools/OmeroWeb/test/integration/test_table.py

```diff
@@ -0,0 +1,164 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for querying OMERO.tables."""
+
+import pytest
+from omeroweb.testlib import IWebTest, get, get_json
+from test_api_projects import get_connection
+
+from omero.grid import WellColumn, StringColumn, DoubleColumn
+
+from django.core.urlresolvers import reverse
+from random import random
+
+
+class TestOmeroTables(IWebTest):
+    """Tests querying of OMERO.table."""
+
+    @pytest.fixture()
+    def user1(self):
+        """Return a new user in a read-annotate group."""
+        group = self.new_group(perms='rwra--')
+        return self.new_client_and_user(group=group)
+
+    @pytest.fixture()
+    def django_client(self, user1):
+        """Return new Django client."""
+        conn = get_connection(user1)
+        user_name = conn.getUser().getName()
+        return self.new_django_client(user_name, user_name)
+
+    @pytest.fixture()
+    def table_data(self):
+        """Return column classes, column names & row data."""
+        col_types = [
+            WellColumn, StringColumn, DoubleColumn, DoubleColumn
+        ]
+        col_names = ["Well", "TestColumn", "SmallNumbers", "BigNumbers"]
+        rows = [
+            [1, 'test', 0.5, 135345.0],
+            [2, 'string', 1.0, 345345.121],
+            [3, 'column', 0.75, 356575.012],
+            [4, 'data', 0.12345, 13579.0],
+            [5, 'five', 0.01, 500.05]
+        ]
+        return (col_types, col_names, rows)
+
+    @pytest.fixture()
+    def omero_table_file(self, user1, table_data):
+        """Create a new OMERO Table and returns the original file ID."""
+        client = user1[0]
+        col_types, col_names, rows = table_data
+
+        columns = []
+        for col_type, name in zip(col_types, col_names):
+            if col_type == StringColumn:
+                columns.append(StringColumn(name, '', 64, []))
+            else:
+                columns.append(col_type(name, '', []))
+
+        tablename = "omero_table_test:%s" % str(random())
+        table = client.sf.sharedResources().newTable(1, tablename)
+        table.initialize(columns)
+
+        data = []
+        for col_type, name, idx in zip(col_types, col_names,
+                                       range(len(col_names))):
+            col_data = [row[idx] for row in rows]
+            if col_type == StringColumn:
+                data.append(StringColumn(name, '', 64, col_data))
+            else:
+                data.append(col_type(name, '', col_data))
+
+        table.addData(data)
+        table.close()
+
+        orig_file = table.getOriginalFile()
```

Comment:
All calls to `getOriginalFile` will need to be moved _before_ `table.close()` Cf. https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-test-integration/lastCompletedBuild/testReport/OmeroWeb.test.integration.test_table/TestOmeroTables/test_table_html/